TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/EditTaskScreen/EditTaskScreenViewController.m

```diff
@@ -0,0 +1,359 @@
+        prioritystr = [dictionaryTask valueForKey:@"prioritystr"];
+
+    }
+
+    status = [NSNumber numberWithUnsignedInteger:1];
+    
+    if ([nid unsignedIntegerValue] > 0){
+        
+        [swNotification setOn:YES];
+        
+    }else{
+        
+        [swNotification setOn:NO];
+
+    }
+    
+    [self formatDueDate];
+    
+    tvTaskTitle.text = title;
+    tvDescription.text = details;
+    
+    scPriority.selectedSegmentIndex = [priority unsignedIntegerValue];
+    
+}
+
+
+- (IBAction)btnDateDueTaskTapped:(id)sender{
+    
+    DateDueScreenViewController *dateDueScreen = [[DateDueScreenViewController alloc] init];
+    
+    [self.navigationController pushViewController:dateDueScreen animated:YES];
+    
+}
+
+
+- (IBAction)scPriorityChanged:(id)sender{
+
+    priority = [NSNumber numberWithInteger:scPriority.selectedSegmentIndex];
+    
+    switch ([priority unsignedIntegerValue]) {
+        case 0:{
+            
+            prioritystr = @"Low";
+            break;
+        }
+        case 1:{
+
+            prioritystr = @"Medium";
+            break;
+        }
+        case 2:{
+    
+            prioritystr = @"High";
+            break;
+        }
+    }
+    
+}
+
+
+-(void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event{
+    
+    [self.view endEditing:YES];
+    
+}
+
+
+-(void) setDateDue:(NSNumber *)dateDueIn{
+    
+    dateDue = dateDueIn;
+    
+}
+
+
+- (IBAction)btnSaveTaskTapped:(id)sender{
+
+    NSLog(@"save task tapped");
+
+    title = tvTaskTitle.text;
+    details = tvDescription.text;
+    
+    if ([title length] == 0){
+        [self showAlertMessage:@"Error!" message:@"Title value is missed"];
+    }
+    
+    
+    if ([details length] == 0){
+        [self showAlertMessage:@"Error!" message:@"Description value is missed"];
+    }
+
+    if ([swNotification isOn] == false){
+        nid = [NSNumber numberWithUnsignedInteger:0];
+    }
+    
+    NSMutableDictionary *dictionaryTask = [[NSMutableDictionary alloc] init];
+    
+    [dictionaryTask setValue:tid forKey:@"tid"];
+    [dictionaryTask setValue:lid forKey:@"lid"];
+    [dictionaryTask setValue:title forKey:@"title"];
+    [dictionaryTask setValue:priority forKey:@"priority"];
+    [dictionaryTask setValue:nid forKey:@"nid"];
+    [dictionaryTask setValue:status forKey:@"status"];
+    [dictionaryTask setValue:dateDue forKey:@"dateto"];
+    [dictionaryTask setValue:details forKey:@"details"];
+    [dictionaryTask setValue:prioritystr forKey:@"prioritystr"];
+
+    TASingleton *singleton = [[TASingleton alloc] init];
+    [singleton addTask:dictionaryTask taskId:lid];
+
+    //if notification date is in the past - do not save notification
+    
+    NSArray *viewControllers = self.navigationController.viewControllers;
+    
+    if ([[viewControllers objectAtIndex:self.navigationController.viewControllers.count - 2] isKindOfClass:[TaskDetailViewController class]] == YES){
```

Comment:
All this wouldn't happen if the controllers exchanged the information in any way other than going up the view stack. Be mindful about data flows not always mapping 1:1 to the navigation flows.